### ISSUE TITLE
[Feature Request] Field-level access policies

### ISSUE THREAD

**Issue by ymc9** [zenstackhq/zenstack#278]
Suggestion from @keanugrievs:

```prisma
hits  Int @default(1)  @override("update", auth().id == user_id && future() == current() + 1)
```

Other thoughts:
```prisma
hits Int @allow('update', future().hits == hits + 1)
```

**Comment by potion-cellar** [comment]
+1 for this

Would like to have some protected fields, for instance we have

```
Organization {
    id                    Int                   @id @default(autoincrement())
    name                  String

    notes                 String?
}
```

where "notes" would be perhaps an internal, non-customer-facing field that only people with a certain role could view. Right now, to accomplish something similar using only zenstack middleware, we need to create a separate "OrganizationNotes" table with a 1-to-1 relationship and put access policies on that table.

**Comment by deiucanta** [comment]
+1 for this

My use case is for `read` control on certain fields. I have a model where some fields should be private - visible only by admins and the author.

**Comment by ymc9** [comment]
Fixed by #638 

https://zenstack.dev/docs/reference/zmodel-language#field-level-policy

**Comment by tlancina** [comment]
@ymc9 would it be possible to get an example of field level policies? I have a similar use case, but can't get them to work on latest. I see the policies are generated, but they don't seem to get called.
